ch.vorburger.osgi.gradle/src/main/java/ch/vorburger/osgi/maven/internal/MavenBuildService.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * ch.vorburger.osgi.gradle
+ *
+ * Copyright (C) 2016 - 2017 Michael Vorburger.ch <mike@vorburger.ch>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ch.vorburger.osgi.maven.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+
+import ch.vorburger.osgi.gradle.internal.BuildService;
+import ch.vorburger.osgi.gradle.internal.BuildServiceListener;
+import ch.vorburger.osgi.gradle.internal.concurrent.ExecutorServiceProviderSingleton;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.Invoker;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+
+/**
+ *
+ */
+public class MavenBuildService implements BuildService {
```

Comment:
FYI I want to look this class over in more details once again when less tired, but I'll send the first review back already; sorry if in 2 rounds and may be more later on this class...